diff --git a/src/js/shCore.js b/src/js/shCore.js
index 19823f1..d554552 100644
--- a/src/js/shCore.js
+++ b/src/js/shCore.js
@@ -92,7 +92,6 @@ var sh = {
 
 	/** Internal 'global' variables. */
 	vars : {
-		discoveredBrushes : null,
 		highlighters : {}
 	},
 
@@ -644,40 +643,39 @@ function alert(str)
  */
 function findBrush(alias, showAlert)
 {
-	var brushes = sh.vars.discoveredBrushes,
-		result = null
-		;
+	var brush = sh.brushes[alias];
 
-	if (brushes == null)
-	{
-		brushes = {};
+  if (brush !== undefined) {
+    return brush;
+  }
 
-		// Find all brushes
-		for (var brush in sh.brushes)
-		{
-			var info = sh.brushes[brush],
-				aliases = info.aliases
-				;
+  // Find brush
+  for (brush in sh.brushes)
+  {
+    var info = sh.brushes[brush],
+      aliases = info.aliases
+      ;
 
-			if (aliases == null)
-				continue;
+    if (aliases == null)
+      continue;
 
-			// keep the brush name
-			info.brushName = brush.toLowerCase();
+    // keep the brush name
+    info.brushName = brush.toLowerCase();
 
-			for (var i = 0, l = aliases.length; i < l; i++)
-				brushes[aliases[i]] = brush;
-		}
-
-		sh.vars.discoveredBrushes = brushes;
-	}
+    for (var i = 0, l = aliases.length; i < l; i++)
+      sh.brushes[aliases[i]] = info;
+  }
 
-	result = sh.brushes[brushes[alias]];
+	brush = sh.brushes[alias];
 
-	if (result == null && showAlert)
-		alert(sh.config.strings.noBrush + alias);
+	if (brush === undefined) {
+    sh.brushes[alias] = null;
+    if (showAlert) {
+      alert(sh.config.strings.noBrush + alias);
+    }
+  }
 
-	return result;
+	return brush;
 };
 
 /**
